#51_Avatar_Customizing

tehokim opened the pull request:
아바타 커스터마이징 기능 구현했습니다. 
AvatarCustomizing Scene에 prefab만 따로 구성이 되어 있습니다.
LobbyScene의 canvases prefab 부분에서 customizeAvatarCanvas를 수정하였습니다.  script는 Assets\Scripts\Lobby\UI\AvatarCustomization 폴더의 CustomizingButtonScript, CustomizingPanelScript, PanelSizeScript 입니다. tabgroupScript도 일부 수정하였습니다. 또한, Assets\Prefabs\UI\Lobby 폴더의 FlexibleColorPicker 에셋 그대로 import 하시면 됩니다. Assets\Resources\texture 폴더 또한 그대로 사용하시면 될 것 같습니다.

tehokim committed:
#51_Avatar_Customizing

#51_Avatar_Customizing

ahnjinmo commented:
pull해보니 Lobby scene에는 씬 내에 오브젝트가 아무것도 없고, AvatarCustomizing scene에서는 Canvases 프리팹이 없다는 오류가 뜹니다. 그리고 아바타 커스터마이징 UI도 동작을 안 하구요! 커밋 단계에서 문제가 있었던 것 같습니다. 다른 분들도 혹시 그런가요?

tehokim committed:
#51_Avatar_Customizing

#51_Avatar_Customizing-re

tehokim commented:
> pull해보니 Lobby scene에는 씬 내에 오브젝트가 아무것도 없고, AvatarCustomizing scene에서는 Canvases 프리팹이 없다는 오류가 뜹니다. 그리고 아바타 커스터마이징 UI도 동작을 안 하구요! 커밋 단계에서 문제가 있었던 것 같습니다. 다른 분들도 혹시 그런가요?

추가 수정해서 다시 commit했는데 또 오류가 있는지 확인 부탁드립니다!

tehokim committed:
#51_Avatar_Customizing_rre

#51_Avatar_Customizing_rre

ahnjinmo commented:
Lobby scene에서 아바타 커스터마이징 정상 작동하는 것 확인했습니다! 
아래 사진처럼 캐릭터 모습이 뒤편에 같이 보이긴 하는데, 이건 나중에 Background 이미지 추가하거나 하면 될 것 같네요!
![dd2](https://user-images.githubusercontent.com/89832796/150112805-ace25697-c705-4f06-bc8a-b9400c85e4bc.jpg)


그리고 AvatarCustomizing scene에서는 TabGroup.cs에서 오류가 떠서 동작을 안하는데 이건 원래 그런 건가요?
![dd제목 없음](https://user-images.githubusercontent.com/89832796/150112482-85d08ea2-4e2b-4d82-88f9-a83652aa9fd2.jpg)

tehokim committed:
#51_Avatar_Customizing_rrre

#51_Avatar_Customizing_rrre

tehokim commented:
AvatarCustomizing scene은 처음에 그냥 프리팹 옮겨두는 용도로 만든거라서 오류가 뜨는게 맞습니다! 그냥 lobby scene을 최종 결과물이라고 생각해주시면 됩니다.
_rrre에 원래 계획 중 하나였던 카메라 줌인과 이동 기능 추가하였습니다.
그리고 카메라 리셋 기능이 있으면 좋을 것 같아서 따로 버튼을 만들어서 추가 구현했습니다.
Assets\Scripts\Lobby\UI\AvatarCustomization 폴더의 AvatarCameraScript입니다.

Hyewon0309 commented:
확인해봤는데 Lobby Scene에서도 TabGroup 관련해서 오류가 있었는데 일단 작동은 잘하네요. 그리고 원래 Done 버튼은 작동하지 않는게 맞나요?

tehokim commented:
넵 저도 그 부분이 의아해서 계속 코드를 봤는데 정확히 어떤 오류인지 잘 모르겠습니다. TabGroup은 네트워크팀 작업을 이어받은 script라서 이거 직접 짜신 분께 한 번 물어봐야 될 것 같아요. 그리고 done 버튼은 신경을 쓰고 있지 않았는데 다시 확인해보겠습니다.

tehokim committed:
0120_오류해결

우선은 색상 변경만 해결했습니다. 사실 텍스쳐도 안되는 것은 아닌데 이건 제가 따로 메터리얼을 만들어서 테스트 해봐야 할 것 같습니다. texture 변경 부분은 추후에 다시 커밋 하겠습니다.

tehokim committed:
0121_수정

* 뒤로 가기 버튼, 컬러 초기화 버튼, material start 초기화 수정
* coding convention 따라서 변수 이름 변경 완료

tehokim commented:
0121_수정사항
1. lobby -> mainroom 넘어갈 때 아바타 커스터마이징 데이터가 안넘어가는 현상 해결
2. avatar material 초기화 문제 해결
3. color picker reset button으로 컬러 피커 색 초기화 기능 구현
4. backbotton으로 customizeAvatarCanvas -> PlayerNameInputCanvas 뒤로가기 기능 구현
5. Assets\Scripts\Lobby\UI\AvatarCustomization 폴더의 모든 script를 code convention에 맞춰서 변수/함수 이름 변경

Hyewon0309 commented:
현재 수정사항은 다 확인했습니다.

한가지 더 수정 부탁드릴 것은 뒤로가기 버튼을 눌러서 닉네임 설정 화면으로 돌아갔다가 다시 커스터마이징으로 돌아오면 색과 캠이 초기화 되어있어야 할 것 같아서 이 부분 확인 부탁드립니다.

tehokim committed:
0123_리셋 수정/카메라 조작 수정

1. 리셋 수정
backbutton시 모든 변수 초기화
colorpicker 자잘한 오류 수정
2. 카메라 조작 수정
카메라 위에 마우스가 있을 때에만 카메라 조정이 되도록 수정

tehokim commented:
0123_수정사항

1. 리셋 수정
- back button시 모든 변수 초기화
- color picker 자잘한 오류 수정
2. 카메라 조정 수정
- 카메라 위에 마우스가 있을 때에만 카메라 조정이 되도록 수정

tehokim committed:
0124_코드수정

0124_코드수정

brainini committed:
[feat] Avatar selection panel caracter move retriction wall

brainini committed:
[feat] Avatar retriction wall

brainini committed:
[merging..] remove old AvatarCustomizingCanvas

brainini committed:
[merging...] Delete Canvas prefab in dev

brainini committed:
Merge branch 'dev' into #51_Avatar_Customizing

brainini committed:
[merge] conflict solved (minimap Player null checking)
